packages/website/src/bin/page-generator/templates/example/index.js

```diff
@@ -12,13 +12,14 @@ const outdent = require('outdent');
  */
 const exampleTemplate = (componentPath, wrapperPath, data = {}) => outdent`
     import React from 'react';
-    import * as Components from '${componentPath}';
+    import dynamic from 'next/dynamic';
     import fileContents from '!!raw-loader!${componentPath}';
-
     import Wrapper from '${wrapperPath}';
-
-    export default () => (
-      <Wrapper data={${JSON.stringify(data)}} fileContents={fileContents}>
+    
+    const DynamicComponent = dynamic(import('${componentPath}')
+    .then(Components => {
+    return () => ( 
+    <Wrapper data={${JSON.stringify(data)}} fileContents={fileContents}>
           {
```

Comment:
Cool, so this is reverted to @pyupew's original solution for this, which I have thoughts and feelings about.

Peter's doing a custom handling of children, expecting children to be an array of objects(?), not a component*. This now works because we are passing an appropriate array.

This is kind of a really bad thing to do with children though - as children are assumed by everything to be a component. We basically create for ourselves a footgun that new devs to the project may stumble over.

There are two ways to resolve this:

1. Don't pass this array as children, and don't pass any children:

```
<Wrapper components={thatArray} />
```

2. Move the logic in (what I ended up doing)

Because demo-driven development, probably going to merge this as-is, but definitely want to come back with with the refactor fairy later.

Reply:
Sure

Reply:
I prefer option 1. The more logic we can avoid including in the templates the better. And also it makes more sense for the example wrapper to contain the render logic for elements like the surrounding border box because they are reused across every example page.